In [ ]:
import os
import torch
from wordcloud import WordCloud
import missingno as msno
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.model_selection import train_test_split

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

folder_path = "/content/drive/MyDrive/DS310/Project/Dataset/datasets/Multi-Label dataset"
single_dataset = "/content/drive/MyDrive/DS310/Project/Dataset/merged_datasets/merged_single_label/merged.csv"

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
all_dfs = []

for file in os.listdir(folder_path):
    if file.endswith(".csv"):
        label = file.replace("_papers.csv", "")
        df = pd.read_csv(os.path.join(folder_path, file))

        df["category"] = label

        all_dfs.append(df)

merged_df = pd.concat(all_dfs, ignore_index=True)
print("Tổng số dòng:", len(merged_df))
merged_df.head()

Tổng số dòng: 12313


,title,link,date_posted,headline,source,related_topics,abstract,full_story,category
0,'Old Smokers' and 'Squalling Newborns' Among H...,https://www.sciencedaily.com/releases/2024/01/...,"January 27, 2024",'Old smokers' and 'squalling newborns' among h...,Royal Astronomical Society,Space & Time,'Hidden' stars including a new type of elderly...,'Hidden' stars including a new type of elderly...,space_time.csv
1,Stars Travel More Slowly at Milky Way's Edge,https://www.sciencedaily.com/releases/2024/01/...,"January 26, 2024",Stars travel more slowly at Milky Way's edge,Massachusetts Institute of Technology,Space & Time,Physicists discovered stars near the edge of t...,By clocking the speed of stars throughout the ...,space_time.csv
2,Astronomers Unravel Mysteries of Planet Format...,https://www.sciencedaily.com/releases/2024/01/...,"January 29, 2024",Astronomers unravel mysteries of planet format...,University of California - Irvine,Space & Time,"TOI-1136, a dwarf star located more than 270 l...",A recently discovered solar system with six co...,space_time.csv
3,Confirmation of Ancient Lake on Mars Builds Ex...,https://www.sciencedaily.com/releases/2024/01/...,"January 26, 2024",Confirmation of ancient lake on Mars builds ex...,University of California - Los Angeles,Space & Time,Ground-penetrating radar on board NASA's Mars ...,"If life ever existed on Mars, the Perseverance...",space_time.csv
4,Cosmic Building Blocks of Life Discovered Thro...,https://www.sciencedaily.com/releases/2024/01/...,"January 29, 2024",Cosmic building blocks of life discovered thro...,University of Münster,Plants & Animals|Space & Time|Fossils & Ruins,Meteorites are fragments of asteroids which fi...,A research team analyses extraterrestrial amin...,space_time.csv


In [ ]:
merged_df["full_story"] = merged_df["full_story"].astype(str).fillna("")
merged_df["abstract"]   = merged_df["abstract"].astype(str).fillna("")

# Lọc bỏ những dòng thực sự rỗng hoặc quá ngắn
merged_df = merged_df[merged_df["full_story"].str.len() > 100]
merged_df = merged_df[merged_df["abstract"].str.len() > 10]
merged_df = merged_df.reset_index(drop=True)

In [ ]:
merged_df_single = pd.read_csv(single_dataset)

# So sánh hai bộ dữ liệu cào ở hai thời điểm khác nhau

In [ ]:
set_all = set(merged_df["link"])
set_single = set(merged_df_single["link"])

common_links = set_all & set_single

In [ ]:
print("Số link merged_df:", len(set_all))
print("Số link merged_df_single:", len(set_single))
print("Số link trùng nhau:", len(common_links))

print("Tỷ lệ trùng so với merged_df:",
      len(common_links) / len(set_all))

print("Tỷ lệ trùng so với merged_df_single:",
      len(common_links) / len(set_single))

Số link merged_df: 8741
Số link merged_df_single: 8698
Số link trùng nhau: 8469
Tỷ lệ trùng so với merged_df: 0.9688822789154559
Tỷ lệ trùng so với merged_df_single: 0.9736721085306967


Nhận xét:
- Dựa trên cột link, kết quả phân tích cho thấy hai tập dữ liệu có mức độ chồng lấp rất cao. Cụ thể, 96.89% số liên kết trong merged_df và 97.37% số liên kết trong merged_df_single là trùng nhau.

Vậy là hai bộ dataset này trùng hợp nhau chỉ khác ở bộ dataset này có thêm một label là related_topic